scopes/pipelines/builder/builder.main.runtime.ts

```diff
@@ -144,51 +147,28 @@ export class BuilderMain {
     aspectName: string,
     name: string
   ): Promise<ArtifactVinyl[]> {
-    const artifactsObjects = this.getArtifactsByExtensionAndName(component, aspectName, name);
+    const artifacts = this.getArtifactsByExtensionAndName(component, aspectName, name);
     const vinyls = await Promise.all(
-      (artifactsObjects || []).map((artifactObject) =>
-        artifactObject.files.getVinylsAndImportIfMissing(component.id._legacy, this.scope.legacyScope)
+      (artifacts || []).map(async (artifact) =>
+        artifact.files.getVinylsAndImportIfMissing(component.id._legacy, this.scope.legacyScope)
```

Comment:
If you change the artifact list to extend array, and all the getArtifactsByXXX to return it.
Then you can move this function to the ArtifactList